[8.0] Add stock_scanner module

Garamotte opened the pull request:
This module allows to write scenarios, to be used (mainly) from smart barcode scanners.
The scenarios can be created directly from Odoo, or added as any other data in a module.
A specific odoo client (written using python/ncurses) is then used to run the scenarios from the barcode scanner (usually through SSH).

A branch for version 9.0 is also available (only an XML ID in the standard stock module changes between 8.0 an 9.0).

Commit:
[IMP] Send False value by default

Commit:
[IMP] Added some debug logs

Commit:
[IMP] Added diagram view on scenario

Commit:
[IMP] Added menus/submenus management + steps stack to be able to go back at any time

Commit:
[IMP] Removed prints

Commit:
[IMP] Updated import/export scripts to take new fields 'type' and 'parent_id' + small bug fixes

Commit:
[FIX] Check for 'parent_id' field existence in import file

Commit:
[IMP] Added missing ondelete caascade

Commit:
[IMP] Added search view for scnarios

Commit:
[FIX] Fixed rewind system

Commit:
[FIX] Fixed background color when rewinding from an error screen

Commit:
[IMP] Fixed views

Commit:
[FIX] Fixed restart and rewind system

Commit:
[IMP] Added type 'shortcut' to launch a same scenario from multiple menus

Commit:
[FIX] Fixed scenarios view for shortcuts

Commit:
[IMP] Added sequence field in transition views

Commit:
[IMP] Removed useless attrs readonly

Commit:
[FIX] Fixed default order for scenarios

Commit:
[FIX] Fixed index out of range error for small lists in sentinel

Commit:
[FIX] Fixed autoconfigure prompt

Commit:
[IMP] Order data in XML file, to avoid random reordering at export

Commit:
[IMP] Lists display can move to show tuncated characters

Commit:
[IMP] Added access rules for hardware

Commit:
[TAG] 1.0

Commit:
[FIX] Fixed unicode displaying error

Commit:
[IMP] Quantity is a float

Commit:
[IMP] Traceback display when an error occurs in a scenario script

Commit:
[FIX] Fixed auto-index for list display

Commit:
[IMP] List steps works with dictionnaries too

Commit:
[IMP] Improve french translation

Commit:
[IMP] Added a new step type to select integer values

Commit:
[IMP] Count steps and transitions at import instead of flooding the screen

Commit:
[FIX] Code was too small

Commit:
[IMP] Read strings char by char to allow catching special keys at any time

Commit:
[IMP] Fixed log text

Commit:
[FIX] Rollback all database changes if an error occurs during the step script

Commit:
[IMP] Back exception is typed, to allow traping several signals

Commit:
[IMP] Added step_back field to check steps that blocks rewind

Commit:
[FIX] Fixed previous steps ids stacking

Commit:
[IMP] Added logfile for tracebacks in sentinel

Commit:
[FIX] Do not crash when typing a long text

Commit:
[IMP] Added a log line when no valid transition is found

Commit:
[IMP] Use %g to format floats instead of str, to avoid having '.0' automatically added

Commit:
[FIX] Do not crash when a list step has no value

Commit:
[FIX] Stop intinite log on error

Commit:
[FIX] Temporary fields must not be required

Commit:
[PEP8] PEP8 Compliant

Commit:
[IMP] Add date and time for tracebacks in logfile

Commit:
[IMP] Added support for default value and max size on text steps

Commit:
[IMP] Auto validate list selections if max character number is reached

Commit:
[FIX] All classes must inherit at least from the core 'object' class

Commit:
[FIX] Auto-validate list only on digit keys

Commit:
[IMP] Transmit message to next step from simple display steps

Commit:
[FIX] Fixed list display when moving right and left

Commit:
[IMP] Added '--directory' option to import and export scripts, default to '.'

Commit:
[IMP] Add an argument to import recursively scenarios from a directory

Commit:
[IMP] Check steps syntax when importing scenario

Commit:
[IMP] Added non-recursion constraint on scenario

Commit:
[IMP] Added a constraint on transition to avoid linking steps of different scenarios

Commit:
[I18N] Updated french translation

Commit:
[IMP] The * key writes a . in quantity selection steps

Commit:
[IMP] The Ctrl+C sequence requests back to the main menu

Commit:
[IMP] Replace full value when pressing the first digit key in quantity selection steps

Commit:
[IMP] Added a method to push values in the keyboard buffer

Commit:
[IMP] Do not display the scenario termination message when returning back to the main menu

Commit:
[IMP] Use standard methods instead of rewriting the same thing two times

Commit:
[IMP] Doubled the size of temporary values in scanner.hardware

Commit:
[IMP] Long texts can be scrolled with Up/Down arrows in all types of steps

Commit:
[IMP] Display the OSV exceptions instead of treating them like unknown exceptions

Commit:
[IMP] Allow to request going back to the previous step on an error message

Commit:
[IMP] Do not quit scenario when returning back to the first step

Commit:
[FIX] Fixed max size in text steps

Commit:
[FIX] Fixed word wrap for steps message display

Commit:
[IMP] Replaced char field by text field

Commit:
[FIX] Fixed cursor position for long texts

Commit:
[IMP] Sentinel colors can now be configured for each terminal in OpenERP

Commit:
[FIX] Catch except_orm exceptions

Commit:
[IMP] Added translations management for the sentinel application + default translation file

Commit:
[I18N] Added french translation for the sentinel application

Commit:
[IMP] Updated default translation

Commit:
[FIX] Fixed encoding error

Commit:
[I18N] Updated french translation

Commit:
[FIX] Fixed encoding issue

Commit:
[IMP] Updated default and french translations + moved the sentinel.pot file

Commit:
[IMP] Add icon on the button

Commit:
[IMP] Cycle menu display

Commit:
[FIX] Fixed import script when we deactivate a scenario in XML file

Commit:
[IMP] Added better names for variables passed to steps exceuted code

Commit:
[IMP] List steps can now have a menu

Commit:
[I18N] Updated fr and en translations

Commit:
[FIX] Fixed new steps default comment

Commit:
[IMP] Added 'O', 'Y' and 'N' shortcuts to select 'Yes' or 'No' in confirm steps

Commit:
[IMP] Added sequence in transitions part of steps view

Commit:
[FIX] Fixed title detection in list steps with dictionnary list

Commit:
[IMP] Use more python magic

Commit:
[IMP] Added list of tuples and list of lists management for sorted list steps with key different than displayed value

Commit:
[FIX] Fixed caret position on list selection step

Commit:
[FIX] Added title management for list of lists and list of tuples selection step

Commit:
[FIX] Flush the input before and after getting a text to avoid capturing extra characters

Commit:
[IMP] Change licence GPL to AGPL

Commit:
[IMP] Add images and description

Commit:
[IMP] Rename image

Commit:
[IMP] Allow to use ~ for home directory

Commit:
[IMP] Add possibility to specify user for execute scenario instead of default user

Commit:
[ADD] Added some demo scenarios

Commit:
[IMP] Added explanation for the steps return types

Commit:
[IMP] Improved module description

Commit:
[IMP] Replaced the warehouses by the OpenERP demo warehouses

Commit:
[FIX] Fixed the demo inventory scenario

Commit:
[IMP] Allow to scan barcodes which contains control characters (<32 ascii)

gurneyalex committed:
[FIX] add invisible scenario_id field in scenario.step form

This prevents a crash when one tries to add a transition from within the scenario.step form

gurneyalex committed:
[REF] wrap long lines in stock_scanner.py

Commit:
Merge pull request #1 from gurneyalex/6.0

[FIX] add invisible scenario_id field in scenario.step form

gurneyalex committed:
[IMP] improve search views of scenario steps and transitions

Added a related field to the scenario itself on the transitions
so that we are able to filter and sort transitions based on the scenario.

gurneyalex committed:
Merge branch '6.0' of git://github.com/syleam/stock_scanner into 6.0

Commit:
Merge pull request #2 from gurneyalex/6.0

6.0

Commit:
[FIX] Use the sys.exc_info() function to follow python 2 standard

Commit:
[ADD] Add README.rst file

Commit:
stock_scanner : automatically retry transactions that failed due to a transient concurrent transaction conflict

Commit:
Adapt patch editor in stock_scanner

Commit:
stock_scanner : automatically retry transactions that failed due to a transient concurrent transaction conflict

odony committed:
stock_scanner: refactor scanner.hardware._scenario_save to completely reset the state when retrying

Commit:
[PEP8] PEP8 Compliant

Commit:
[IMP] Migration to v7.0

Commit:
Merge branch '6.0' into 7.0

Commit:
[FIX] Remove scenario_id from transition on export

Commit:
[FIX] Remove scenario_id from transition on export

Commit:
[FIX] Fixed cursor position, caused crash with large number of lines (tried to display the cursor out of the screen)

Commit:
Merge branch '6.0' into 7.0

Commit:
[IMP] Delete old steps and transitions from scenarios on import

Commit:
Merge branch '6.0' into 7.0

Commit:
[ADD] Add a simple barcode decode scenario, useful for barcode tests

Commit:
Merge branch '6.0' into 7.0

Commit:
[IMP] Fix copy_data arguments

Commit:
[IMP] Add company_id on scenario

Commit:
[IMP] Manage title display for all step types

Commit:
[IMP] Store scenario id and name at scenario change

Commit:
[IMP] Display scenario name if no title is defined by the step

Commit:
[IMP] Add scenario name after id in error log

Commit:
[FIX] Fixed scenario_name values

Commit:
[IMP] Used title for steps description scenario

Commit:
[IMP] Add test file management in the sentinel.py client

Commit:
[FIX] Return to the main menu at initialization when using a test file

lmignon committed:
Add a new flag on a step to prevent back action on a specific step

lmignon committed:
[FIX] avoid replaying the last transition when 'ESC' is pressed.

lmignon committed:
[FIX] store the current step id in the previous step list right after a transition completion from this step

Commit:
[IMP] Add step return code A, for automatic steps (the next step will be automatically called, without being sent to the client)

Commit:
[IMP] Updated README file with step types, to explain the new automatic step type

Commit:
[IMP] Removed unused imports + PEP8 Compliant

Commit:
[IMP] Add a note about openobject-library

Commit:
[PEP8] PEP8 Compliant

Commit:
[PEP8] PEP8 Compliant

damdam-s committed:
[IMP] migration v8

damdam-s committed:
[FIX] error when height overflow

damdam-s committed:
[IMP] access rights now loaded

lmignon committed:
PEP8 The Odoo part of this addons
FIX use the user declared in the termina to execute the scanner.call
IMP refactor duplicated code and remove unused function

lmignon committed:
[ADD] new field user_ids and group_ids on scanner.scenario

lmignon committed:
Implement timeout,
Define security rules
Add configuration wizard

lmignon committed:
[FIX] Use the proper decorator on methods
Update tests scenarii

lmignon committed:
[IMP] Implement login/logout scenarii

lmignon committed:
[IMP] Add Credits section in Readme.rst and remove multiple copyrights others than Syleam

lmignon committed:
[IMP]Add copyright header on the wizard

lmignon committed:
[ADD]Travis tests runner

lmignon committed:
[IMP] Add automated tests

lmignon committed:
[FIX] tests

lmignon committed:
[IMP] travis Migrating from legacy to container-based infrastructur

lmignon committed:
[PYLINT] remove print statement

lmignon committed:
[PEP8]

lmignon committed:
[PEP8]

lmignon committed:
[ADD] Test security and login/logout

lmignon committed:
[FIX] Regression introduced when fixing pep8 errors

lmignon committed:
[ADD] badges

lmignon committed:
[IMP] Add documentation

lmignon committed:
Remove badge

Commit:
export_scenario : remove some unused keys

Commit:
export_scenario : remove some unused keys

Commit:
improve import/export of group_ids and user_ids

lmignon committed:
Merge pull request #1 from acsone/8.0-improved-port-lga

export_scenario : remove some unused keys

lmignon committed:
Merge pull request #2 from acsone/8.0-login_logout-lga

mprove import/export of group_ids and user_ids

Commit:
Move login/logout configuration wizard into Warehouse settings

lmignon committed:
Merge pull request #3 from acsone/8.0-login_logout-lga

Move login/logout configuration wizard into Warehouse settings

Commit:
Fix typo

lmignon committed:
Merge pull request #4 from acsone/8.0-login_logout-lga

Fix typo (+ correct test)

Commit:
Improve configuration view

lmignon committed:
Merge pull request #5 from acsone/8.0-login_logout-lga

Improve configuration view

Commit:
Merge remote-tracking branch 'github/acsone/8.0-login_logout-lmi' into 8.0

Commit:
[IMP] Improved v8.0 migration

Commit:
[REM] Remove semaphores from scenario execution (was already broken, and don't work at all with workers mode)

Commit:
[IMP] Add scenarios autoimport on module update

Commit:
[FIX] Readd missing data in demo file + fixed tests for last changes (more demo data)

Commit:
[PEP8] Flake8 validation (not ignoring E501)

Commit:
[IMP] Updated export script to generate a Scenario_Name.scenario file instead of scenario.xml

Commit:
[IMP] Updated README.rst file

Commit:
[IMP] Follow OCA Guidelines

Commit:
[IMP] Remove refrence_res_id fields

UUIDs were not easy to use when editing python files.
We now use model, scenario, step and transition names to generate the
XML ID instead of autogenerated XML IDs from UUID.

Commit:
[FIX] Split models one per file

Commit:
[IMP] Export groups XML IDs, and fallback to xml_id of the group value has not been found as name (for retro-compatibility with old scenario files)

Commit:
[IMP] Exported Login and Logout scenarios as scenario files

Commit:
[FIX] Add noupdate management in scenarios + added this attribute on Login and Logout scenarios

Commit:
[FIX] Reset hardware values before launching the tests

Commit:
[FIX] Add missing contributor in README file

Commit:
[IMP] Replaced the concatenated strings by a real model for steps history (data are more consistent)

Commit:
[FIX] Prevent returning to the previous step when reconnecting after a disconnect/crash of the sentinel

Commit:
[FIX] Return back to the previous step after a 'No value available' error on the list step, instead of going to the next step with a False value

Commit:
[FIX] FIxed comparison between scenario step and id of a scenario step

Commit:
[FIX] Added missing argument when executing automatic steps

Commit:
[FIX] Return a float value instead of string for _select_quantity in sentinel.py

Commit:
[FIX] Fixed pot path in init command + added update command

Commit:
[IMP] Updated translations

Commit:
[FIX] Fixed broken translations without breaking pep8

Commit:
[IMP] Make scenarios code translatable

Commit:
[FIX] Add forgotten context from sentinel to allow scenarios translation

Commit:
[IMP] Make scenarios name translatable : all client-side displayed values are now translatable

Commit:
[FIX] Typos

Commit:
[IMP] Updated login/logout scenarios to be translated

Commit:
[IMP] Updated demo scenarios to be translated

Commit:
[I18N] Improved french translation

Commit:
[PEP8] PEP8 Compliant

Commit:
[IMP] Added missing eval function in README file

Commit:
[IMP] The size argument is no longer required on fields.Char and has been removed (no need to limit the size, specially on condition, which can be very long)

Commit:
[IMP] Moved python code syntax check into the step model, as constraint
[IMP] Added syntax check for python condition on transition

Commit:
[FIX] Default value must fit the good type

Commit:
[FIX] Store the used transition in history to keep the tracer when returning back

Commit:
[FIX] Use only ids to avoid need of isinstance hacks

Commit:
[FIX] Fixed old API to new API migration

Commit:
[FIX] Fixed typo

Commit:
[FIX] Fixed Readme

Commit:
[FIX] Added missing access rights

Commit:
[FIX] Remove check for removed field in tests

Commit:
[FIX] Store the transition in history when going to the step, not when quitting the step, to avoid returning back to the next step

Commit:
[FIX] Send the password for the password step in test, even if it's the same value as login

Commit:
[REM] Removed useless files in the stock_scanner directory

elicoidal commented on stock_scanner/.description:
```diff
@@ -0,0 +1 @@
+Allows managing barcode readers with simple scenarios
```
remove file?

elicoidal commented on stock_scanner/.oerp_sentinelrc:
```diff
@@ -0,0 +1,5 @@
+[openerp]
```
remove file?

elicoidal commented on stock_scanner/README.rst:
```diff
@@ -0,0 +1,187 @@
+====================================================
```
Hi @sylvain-garancher Thanks for your contribution
Please follow the OCA README template:
https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

elicoidal commented on stock_scanner/README.rst:
```diff
@@ -0,0 +1,187 @@
+====================================================
+
+Allows managing barcode readers with simple scenarios
+- You can define a workfow for each object (stock picking, inventory, sale, etc)
```
rst list

elicoidal commented on stock_scanner/README.rst:
```diff
@@ -0,0 +1,187 @@
+Stock Scanner : WorkFlow engine for scanner hardware
+====================================================
+
+Allows managing barcode readers with simple scenarios
```
colon

elicoidal commented on stock_scanner/README.rst:
```diff
@@ -0,0 +1,187 @@
+- You can define a workfow for each object (stock picking, inventory, sale, etc)
+- Works with all scanner hardware model (just SSH client required)
+
+The "sentinel" specific ncurses client, available in the "hardware" directory, requires the "openobject-library" python module, available from pip :
```
move to installation section

elicoidal commented on stock_scanner/README.rst:
```diff
@@ -0,0 +1,187 @@
+Some demo/tutorial scenarios are available in the "demo" directory of the module.
+These scenarios, are automatically imported when installing a new database with demo data.
+
+How does it work ?
```
Usage

elicoidal commented on stock_scanner/README.rst:
```diff
@@ -0,0 +1,187 @@
+==================
+
+On start-up, the client lists available scenarii.
+When the user selects a scenario, the current scenario and step are stored on the hardware configuration's entry in Odoo.
```
s/step/steps

elicoidal commented on stock_scanner/README.rst:
```diff
@@ -0,0 +1,187 @@
+When the user selects a scenario, the current scenario and step are stored on the hardware configuration's entry in Odoo.
+
+When the client sends a message to the server, the next step is selected depending on the current step and the message sent.
+Then, the server returns the result of the step, which contains the step type code and the text to display on the hardware screen.
```
The server then returns the result of the step,[..] its type code

elicoidal commented on stock_scanner/README.rst:
```diff
@@ -0,0 +1,187 @@
+
+If needed enable Login/Logout
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The module come with 2 predifined scenarii for Login and Logout. The functionality is disabled by default and the user to use in
```
fix all text following PEP8

elicoidal commented on stock_scanner/README.rst:
```diff
@@ -0,0 +1,187 @@
+    * Create a *Technical User* 'sentinel' **without roles in Human Resources** and with 'Sentinel: technical users' checked.
+    * Use this user to launch your sentinel session.
+
+Be carefull, the role *Sentinel: technical users* is a technical role and should be used only by sentinel.
```
careful
should only be used

elicoidal commented on stock_scanner/README.rst:
```diff
@@ -0,0 +1,187 @@
+
+Be carefull, the role *Sentinel: technical users* is a technical role and should be used only by sentinel.
+
+The timeout of sessions is managed by a dedicated cron that reset the incative sessions. The timeout can be configured on 
```
indicative

elicoidal commented on stock_scanner/__init__.py:
```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
```
Please use simple headers following OCA templates
https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3

elicoidal commented on stock_scanner/__openerp__.py:
```diff
@@ -0,0 +1,70 @@
+#
+##############################################################################
+
+{
```
Please follow the OCA README template:
https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst
Add OCA author
Version 8.0.1.0.0
Remove empty keys

elicoidal commented on stock_scanner/hardware/sentinel.py:
```diff
@@ -0,0 +1,888 @@
+    pass
+
+if __name__ == '__main__':
+    curses.wrapper(Sentinel)
```
Update images to v8

elicoidal commented on stock_scanner/models/scanner_scenario_transition.py:
```diff
@@ -0,0 +1,133 @@
+        string='From',
+        required=True,
+        ondelete='cascade',
+        help='Step which launches this transition')
```
Add periods at the end of all help sentences

elicoidal commented on stock_scanner/scripts/export_scenario.py:
```diff
@@ -0,0 +1,259 @@
+##############################################################################
+
+import re
+from oobjlib.connection import Connection
```
shouldnot we add exception handling for the import?

elicoidal commented:
That's an impressive contribution!
I wish all modules were that documented :)

Commit:
[IMP] Updated README.rst file following OCA guidelines

Commit:
[IMP] Follow OCA guidelines for each file licence + add period on help sentences

Garamotte commented:
Thanks for this first review Eric.
I updated `README.rst` and all licences in files to follow OCA guidelines and fix typos.
Periods have been added on all help sentences.
The `.description` file has been removed (included in the `summary` key in `__openerp__.py`).
The `.oerp_sentinelrc` file has been removed (included in the relevant part of `README.rst`.

laetitia-gangloff commented:
:+1:  (functional)

We use this module since 7.0. And we have also a version 8.0 in production. All works as expected.
Thank you !

elicoidal commented:
Great! Thanks again for your contribution

clonedagain commented:
Maybe you could leave the .mo file out. They're complied from .po files I suppose?

clonedagain commented on stock_scanner/hardware/sentinel.py:
```diff
@@ -0,0 +1,866 @@
+        """
+        # Read user configuration
+        config = ConfigParser.SafeConfigParser(DEFAULT_CONFIG)
+        config.read('.oerp_sentinelrc')
```
Maybe it could be named `.openerp_sentinelrc`, given that Odoo still uses `.openerp_serverrc`?

Garamotte replied:
Maybe add a second possible name, to avoid breaking already in production instances ?

Commit:
[IMP] Removed vim comments (pointed out by OCA lint check)

lmignon commented:
:+1:  We use this module for many of our clients. Through this, they are able to perform all procedures for the management of a stock through scanner hardware (ex Motorala MC3190 winCE, MC3200 android). Moreover the approach of emulating and controlling  the user interface in a terminal eases the deployments because once the scanner hardware are configured for ssh access to a terminal running 'sentinel' there is no more need to deploy something on the scanner hardware. For the record, our client is able to make a full inventory into 2 warehouses with more than 10.000 locations in less than a few hours. The inventory is done by 30 employees each equipped with a scanner and operating in parallel.

clonedagain commented on stock_scanner/hardware/datalogic/PowerScan/PowerscanService.py:
```diff
@@ -0,0 +1,219 @@
+##closeserial()
+##exit()
+##x.greenledon()
+##x.cursorcr()
```
Please remove this, it's in git history.

clonedagain commented on stock_scanner/hardware/datalogic/PowerScan/PowerscanService.py:
```diff
@@ -0,0 +1,219 @@
+    pws.write("-> %s" % msg)
+    pws.cr()
+    menu = ['Inventaire', 'Reception', 'Preparation', 'Controle', 'Terminer']
+    #displaymenu(PS,menu,0,0)
```
Please remove this

clonedagain commented on stock_scanner/hardware/datalogic/PowerScan/PowerscanService.py:
```diff
@@ -0,0 +1,219 @@
+while True:
+    time.sleep(1)
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
OCA doesn't use those I think

clonedagain commented on stock_scanner/hardware/datalogic/PowerScan/hardware.py:
```diff
@@ -0,0 +1,531 @@
+    sys.stderr.write('User ask to shutdown the program')
+    sys.exit(0)
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
OCA doesn't use those I think

clonedagain commented on stock_scanner/hardware/datalogic/PowerScan/powerscan.py:
```diff
@@ -0,0 +1,235 @@
+
+
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
OCA doesn't use those I think

clonedagain commented on stock_scanner/load_scenario.py:
```diff
@@ -0,0 +1,235 @@
+openerp.tools.convert_file = scenario_convert_file
+openerp.tools.convert.convert_file = scenario_convert_file
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
OCA doesn't use those I think

clonedagain commented on stock_scanner/models/scanner_hardware_step_history.py:
```diff
@@ -0,0 +1,26 @@
+        string='Message',
+        help='Message sent during execution of the step.')
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
OCA doesn't use those I think

clonedagain commented:
What is the `datalogic` directory for please?
Is it some kind of hardware driver for a specific brand or model?

Garamotte commented:
The `datalogic` directory contains a client for a specific brand/model.
It was developped for a customer using the very first version of this module (on OpenERP v5.0), and wasn't updated since 2011, but I kept the files here, because it can be helpful for someone, as an example.

I removed the vim comments (travis lint check pointed this out too).

clonedagain commented:
@sylvain-garancher then I think you should drop the `datalogic` directory altogether.
If it's still compatible with the current version, you should spin it off to another module so that interested users can find it.
Otherwise just remove it please, and add a note in the README with the git hash.

Commit:
[I18N] Updated french translation

Commit:
[REM] Removed the hardware/datalogic directory, leaving a link into the README.rst file

Garamotte commented:
Keeping the git hash in `README.rst` is  good idea, and it's enough to keep access to the files.
I dropped the `datalogic` directory and added a link to the directory at the current commit on GitHub.

French translation of the module updated (help sentences with periods + two forgotten constraints strings).

For the `.mo` files, I suppose you are talking about `sentinel.py` translation.
They are compiled from the `.po` files indeed, with this very simple script : https://github.com/syleam/stock-logistics-workflow/blob/8.0-add-stock_scanner-module/stock_scanner/hardware/i18n/translate.sh
I choosed to commit the `.mo` files, as it eases updates : one just has to `git pull` when updating the module, without having to think about compiling new translations.

clonedagain commented:
@sylvain-garancher I see your point. We usually don't commit both the compiled/generated objects and their sources, because eventually they drift apart.
Maybe it would be acceptable if you provided a git pre-commit hook?
Another solution would be to generate the PO files at runtime. It could be as simple as running the script from the main program (but that would make it unix-only I'm afraid).

Commit:
[IMP] Replaced v6.0 GTK client screenshots by v8.0 web client screenshots

Commit:
[IMP] Add .openerp_sentinelrc as name for the configuration file, to follow standard .openerp_serverrc name

Garamotte commented:
The link for the `datalogic` directory has been fixed to point to the OCA repository.

The images are now v8.0 screenshots, instead of v6.0 GTK client screenshots.

The `sentinel.py` client now accepts `.oerp_sentinelrc` or `.openerp_sentinelrc` configuration files.
If the two are available, `.openerp_sentinelrc` is loaded after, so its values overwrite others.

I think the pre-commit hook is the best solution if we have to remove the `.mo` files from the repository, but it can be forgotten too.
Does anyone else have an opinion about the `.mo` files ?

Commit:
[FIX] Fixed display of the last line when scrolling long texts

Commit:
[FIX] Fixed error handling : Remove unnecessary try..except

This try..except was originally a try..finally statement, used to be
sure the semaphore was always released.

Commit:
[FIX] Fixed the inventory demo scenario

Commit:
[FIX] Fixed the location stock display scenario

gurneyalex commented:
I'm favorable to a solution where .mo files are not commited to the repository. 

Thanks for contributing this module. I deployed a version 2 years ago and it was already great at the time.

clonedagain commented:
Maybe you could simply generate the MO if it's not found? or simply throw an error with the directions to generate them? I'm not sure how difficult or reliable that would be.

Commit:
[REM] Removed useful demo scenarios (moved in separate modules as real data)

Commit:
[IMP] Change default transition type to keyboard

Commit:
[FIX] Fixed PEP8 issue

Commit:
[FIX] Fixed fields display

damdam-s commented:
What about doing the sentinel part in a `pip` package on which odoo module `stock_scanner` depends.
So the translation of this app (mo files) could be generated by the install script of the setup.

Commit:
[PEP8] Fixed PEP8 issue

Commit:
[SEC] Replace eval by safe_eval

Commit:
Add 'stock_scanner/' from commit '607ce958c3d29d4202df677bc64009d9e4869d61'

git-subtree-dir: stock_scanner
git-subtree-mainline: 29ae6221830ebf9b9fd4ade5550309c1cdeb0c95
git-subtree-split: 607ce958c3d29d4202df677bc64009d9e4869d61

Garamotte commented:
For the sentinel translation files issue, I think @damdam-s idea is good, as the sentinel is not related to a specific Odoo version. If we separate the sentinel.py client in its own pip-installable python module, I think the sentinel should be moved in another git repo.
I don't know how to build mo files during install using pip though.

As this PR has not been merged yet, I took the liberty to rebase it on our module's own repository (where I added the changes made here + some other fixes).

I also removed the useful scenarii that were available in the "demo" directory, to split them in separate modules. Even if scenarii are made to be custom, it could be great to have a set of modules, each providing some "standard" scenarii.
Maybe this could become a new repository, grouping all scenarii modules ? Probably including the stock_scanner module itself ?

gurneyalex commented:
@sylvain-garancher that's a great idea. 

Regarding the generation of the `.mo` files, I think this can be done in `setup.py`. See for example https://wiki.maemo.org/Internationalize_a_Python_application#Include_translations_in_your_installation

gurneyalex commented:
@sylvain-garancher I'm merging this PR. We can improve it later.
